### PR TITLE
Reserve proofs rebased

### PR DESCRIPTION
**Pull request by xmrdsc**
Fixes #1508 

- Fixed some layout issues/typos. 
- Implemented stoffu's suggestions

### CODE REVIEW

**Comment by xmrdsc** [comment]
@stoffu Mind taking a look if you have time? :-)

File: main.qml [hunk 1]
```diff
@@ -110,6 +110,8 @@ ApplicationWindow {
         if(seq === "Ctrl+S") middlePanel.state = "Transfer"
         else if(seq === "Ctrl+R") middlePanel.state = "Receive"
         else if(seq === "Ctrl+K") middlePanel.state = "TxKey"
+        else if(seq === "Ctrl+W") middlePanel.state = "ReserveProof"
+        else if(seq === "Ctrl+S") middlePanel.state = "SharedRingDB"
```
Comment (mmbyday) [inline main.qml]:
```suggestion
```

Duplicate of SharedRingDB

File: pages/ReserveProof.qml [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        MoneroComponents.LineEdit {
+            id: amountLine
+            Layout.fillWidth: true
+
+            labelText: qsTr("Amount") + translationManager.emptyString
+            placeholderText: qsTr("Amount to prove") + translationManager.emptyString
+            placeholderFontSize: 16 * scaleRatio
+            inlineButtonText: qsTr("All") + translationManager.emptyString
+            inlineButton.onClicked: amountLine.text = "(all)"
+            enabled: amountLine.text === "(all)" || amountLine.text > 0
```
Comment (mmbyday) [inline pages/ReserveProof.qml]: Currently does not work. Perhaps re-use the validator from the transfer page.

File: pages/ReserveProof.qml [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        }
+
+        MoneroComponents.LineEdit {
+            id: getMessageLine
+            Layout.fillWidth: true
+
+            labelText: qsTr("Message") + translationManager.emptyString
+            placeholderFontSize: 16 * scaleRatio
+            fontSize: 16 * scaleRatio
+            placeholderText: qsTr("Message") + translationManager.emptyString;
```
Comment (mmbyday) [inline pages/ReserveProof.qml]:
```suggestion
            placeholderText: qsTr("Optional message") + translationManager.emptyString;
```

Consistency with the check reserve proof field.

File: pages/ReserveProof.qml [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+            readOnly: false
+            copyButton: true
```
Comment (mmbyday) [inline pages/ReserveProof.qml]: paste would be more practical, imo

File: pages/ReserveProof.qml [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+        MoneroComponents.LineEdit {
+            id: proveMessageLine
+            Layout.fillWidth: true
+
+            labelText: qsTr("Message") + translationManager.emptyString
+            placeholderFontSize: 16 * scaleRatio
+            fontSize: 16 * scaleRatio
+            placeholderText: qsTr("Optional message") + translationManager.emptyString;
+            readOnly: false
+            copyButton: true
```
Comment (mmbyday) [inline pages/ReserveProof.qml]: Paste would be more practical, IMO :).

File: pages/ReserveProof.qml [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        MoneroComponents.LineEdit {
+            id: signatureLine
+            Layout.fillWidth: true
+
+            labelText: qsTr("Signature") + translationManager.emptyString
+            placeholderText: qsTr("Proof to verify") + translationManager.emptyString;
+            placeholderFontSize: 16 * scaleRatio
+            fontSize: 16 * scaleRatio
+            readOnly: false
+            copyButton: true
```
Comment (mmbyday) [inline pages/ReserveProof.qml]: Paste would be more practical, IMO :).

File: pages/ReserveProof.qml [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        MoneroComponents.LineEdit {
+            id: addressLine
+            Layout.fillWidth: true
+
+            labelText: qsTr("Address") + translationManager.emptyString
+            placeholderFontSize: 16 * scaleRatio
+            fontSize: 16 * scaleRatio
+            placeholderText: qsTr("Address") + translationManager.emptyString;
+            readOnly: false
+            copyButton: true
```
Comment (mmbyday) [inline pages/ReserveProof.qml]: Paste would be more practical, IMO :).

File: pages/ReserveProof.qml [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+            anchors.bottomMargin: 3 * scaleRatio
+        }
+
+        MoneroComponents.Label {
+            fontSize: 24 * scaleRatio
+            text: qsTr("Check Reserve Proof") + translationManager.emptyString
+        }
+
+        Text {
+            text: qsTr("Verify the validity of a reserve proof") + translationManager.emptyString
```
Comment (mmbyday) [inline pages/ReserveProof.qml]:
```suggestion
            text: qsTr("Verify the validity of a reserve proof.") + translationManager.emptyString
```

File: pages/ReserveProof.qml [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+        spacing: 20 * scaleRatio
+
+        MoneroComponents.Label {
+            fontSize: 24 * scaleRatio
+            text: qsTr("Get Reserve Proof") + translationManager.emptyString
+        }
+
+        Text {
+            text: qsTr("This page allows you to interact with the shared ring database. " +
+                       "This database is meant for use by Monero wallets as well as wallets from Monero clones which reuse the Monero keys.") + translationManager.emptyString
```
Comment (stoffu) [inline pages/ReserveProof.qml]: This description is completely off.

**Comment by jonathancross** [comment]
Ping @xmrdsc

**Comment by xmrdsc** [comment]
Old PR is old.

**Pull request closed without merging**